test/reentrancy/ReentrancyTest.t.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import { Test } from "forge-std/Test.sol";
+import { Deploy } from "../../script/Deploy.s.sol";
+import { DeploymentConfig } from "../../script/DeploymentConfig.s.sol";
+import { AttackController } from "./AttackController.sol";
+import { AttackAccount } from "./AttackAccount.sol";
+import { MiniMeToken } from "../../contracts/MiniMeToken.sol";
+import "forge-std/console.sol";
+
+error NotEnoughBalance();
+
+contract ReentrancyTest is Test {
+    AttackController internal attackController;
+    AttackAccount internal attackAccount;
+    MiniMeToken internal minimeToken;
+    DeploymentConfig internal deploymentConfig;
+    address internal deployer;
+    address internal attackerEOA = makeAddr("attackerEOA");
+
+    event Transfer(address from, address to, uint256 value);
+
+    function setUp() public {
+        Deploy deployment = new Deploy();
+        (deploymentConfig,, minimeToken) = deployment.run();
+        (deployer,,,,,,) = deploymentConfig.activeNetworkConfig();
+
+        vm.prank(attackerEOA);
+        attackAccount = new AttackAccount(minimeToken);
+        attackController = new AttackController(attackerEOA, attackAccount);
+    }
+
+    function testAttack() public {
+        address sender = makeAddr("sender");
+        address receiver = address(attackAccount);
+
+        uint256 fundsAmount = 10_000;
+        uint256 allowanceAmount = fundsAmount * 6;
+        uint256 sendAmount = fundsAmount;
+
+        // ensure `sender` has funds
+        vm.prank(deployer);
+        minimeToken.generateTokens(sender, fundsAmount);
+
+        // change controller to AttackController
+        vm.prank(deployer);
+        minimeToken.changeController(payable(address(attackController)));
+
+        // sender sends tokens to receiver
+        vm.startPrank(sender);
+        minimeToken.approve(receiver, allowanceAmount);
+
+        vm.startPrank(attackerEOA);
+
+        try attackAccount.attack(sender, receiver, sendAmount) {
+            console.logString("attack succeeded");
+        } catch (bytes memory err) {
+            if (keccak256(abi.encodeWithSignature("NotEnoughBalance()")) == keccak256(err)) {
```

Comment:
I think what you're looking for here would be `NotEnoughBalance.selector`.

But overall, I think we can remove the `try/catch` altogether here now and just call `attackerAccount.attack()` given that we have assertions coming after this?

Reply:
This was needed before I made some changes in the attack controller. Before the change it would throw, and I was looking for a specific throw. 
Now it does not throw anymore, it would just do nothing.